client/src/components/SendMessage.js

```diff
@@ -36,36 +39,73 @@ class SendMessage extends React.Component {
     return;
   }
 
+  hideEmojiPicker() {
+      this.setState({
+          ...this.state,
+          emojiPickerActive : false
+      });
+  }
+  onKeyUp(event) {
+      if (this.state.emojiPickerActive){
+          if (event.which === 9 || event.which === 37 || event.which === 39) {
+              // Cycle through preview
+              this.refs.emojipicker.onKeyUp(event);
+          } else {
+              const lastWord = this.refs.message.value.split(" ").pop();
+              this.refs.emojipicker.filterWord(lastWord);
+          }
+      }
+  }
+
   onKeyDown(event) {
     // console.log("KEYDOWN", event.type, event.which);
+    if (this.state.emojiPickerActive) {
+        this.refs.emojipicker.onKeyDown(event);
+    }
     if(event.which === 9) {
       // Tab
       event.preventDefault();
       // TODO: autocomplete user names
     } else if(event.which === 186) {
       // ':'
-      // TODO: bring up emoji preview
-    }
+      if (this.props.useEmojis) {
+         this.setState({
+             ...this.state,
+             emojiPickerActive : true
+         });
+      }
+  }
     return;
   }
+  displayEmojiText (emojiText) {
+      let text = this.refs.message.value.split(" ");
+      text.pop();
+      text.push(emojiText);
+      this.refs.message.value = text.join(" ");
+  }
 
   render() {
     return (
       <div className="SendMessage">
         <form onSubmit={this.sendMessage.bind(this)}>
+            {this.props.useEmojis &&
```

Comment:
I prefer putting conditional to variables in JSX. So instead of inlining the code, you could do the following at the beginning of the render function:
```javascript
const emojiPicker = this.props.useEmojis && this.state.emojiPickerActive ?
  <EmojiPicker ref='emojipicker'
    hideEmojiPicker ={this. hideEmojiPicker.bind(this)}
    displayEmojiText ={this. displayEmojiText.bind(this)}
  /> : null
```

and include it in the `<form>` with:
```javascript
<form onSubmit={this.sendMessage.bind(this)}>
  {emojiPicker}
  <input
  ...
```